### PR TITLE
Fixes #143: Added support for BI_BITFIELDS BMP format.

### LINKED ISSUE #143: Add support for loading 32-bit BMPs with BI_BITFIELDS compression
The current `TCustomBitmap32.LoadFromStream` implementation only supports `BI_RGB` compression.

The task of this issue is to extend this with support for the `BI_BITFIELDS` compression type.

This means that for [V1 headers](https://docs.microsoft.com/en-us/previous-versions/dd183376(v=vs.85)) with `BI_BITFIELDS` compression we need to take the color table into account and for [V4+ headers](https://docs.microsoft.com/da-dk/windows/win32/api/wingdi/ns-wingdi-bitmapv4header?redirectedfrom=MSDN) with `BI_BITFIELDS` compression we need to take the color mask into account.
The V2 and V3 headers are undocumented (but the layout is known) so we can ignore those.

The solution should be verified with the bitmaps of the [BMP Suite](https://github.com/jsummers/bmpsuite).

_Originally posted by @andersmelander in https://github.com/graphics32/graphics32/issue_comments/705610626_

### PR DESCRIPTION
**Pull request by andersmelander**
- LoadFromStream has been almost completely rewritten.
- LoadFromStream now supports both BI_RGB and BI_BITFIELDS "compression".
- The BMP and DIB part has been separated so the DIB part can be reused by LoadFromResource*.
- LoadFromResource* now uses the new DIB reader instead of just deferring to TBitmap.
- SaveToStream has been modified to use the proper header structure declarations.
- The DIB part of SaveToStream has been split into a separate method.

Implementation has been tested with the 32-bit RGB and RGBA bitmaps of the [BMP suite](https://github.com/jsummers/bmpsuite). Only the "Good" and the "Questionable" bitmaps were tested.

Known issues:
- For BI_BITFIELDS with less or more than 8 bits used per channel (for example 2 bit alpha) the decoder does not scale the channel values to the 0..255 range. The correct behavior isn't documented but it seems MSPAINT and Windows Explorer either performs some sort of scaling or simply ignores the alpha values if it's using less than 8 bits.
- The RGBA_FORMAT conditional define is only partially handled for BI_RGB compression as the R and B channels aren't swapped as they should be. Not a big problem as the old implementation didn't handle RGBA_FORMAT at all. For BI_BITFIELDS compression RGBA_FORMAT is fully handled.

### CODE REVIEW

**Review by CWBudde — APPROVED** [review]
I jut had a brief look. It seems quite sophisticated and if it works with the test set, I'm OK with it.

**Comment by andersmelander** [comment]
> It seems quite sophisticated 

It's a bit more complex than what I would have liked but less than what it could have been had I not opted to leave out some of the possible performance optimizations. Since this isn't performance critical code I think it's more important to keep the code readable. I've also only included recovery handling for the most basic invalid bitmap cases.

I'd like to wait for some feedback from @dslutej before I merge this.

**Comment by dslutej** [comment]
@andersmelander, works nicely. 👍 

Excellent work, I've skimmed through the changes and stepped through code while loading our pics and everything looks perfect. 

As for performance, by far the biggest gain I have seen is changing:

**Pull request merged**
